source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -303,6 +303,31 @@ const std::string& SslSocket::urlEncodedPemEncodedPeerCertificate() const {
   return cached_url_encoded_pem_encoded_peer_certificate_;
 }
 
+const std::string& SslSocket::urlEncodedPemEncodedPeerCertificateChain() const {
+  if (!cached_url_encoded_pem_encoded_peer_cert_chain_.empty()) {
+    return cached_url_encoded_pem_encoded_peer_cert_chain_;
+  }
+
+  STACK_OF(X509)* cert_chain = SSL_get_peer_cert_chain(ssl_.get());
```

Comment:
Oops, one more thing:
```
if (cert_chain == nullptr) {
    return cached_url_encoded_pem_encoded_peer_cert_chain_;
}
```

Reply:
I believe `sk_X509_num(cert_chain)` will return `-1` if `cert_chain` is `NULL`. Still, I can add the explicit check if needed.

Reply:
Addressed in f4c974d.